infrastructure/azure.tf

```diff
@@ -328,6 +354,32 @@ resource "azurerm_app_service_plan" "azurerm_app_service_plan" {
     # }
 }
 
+## FUNCTIONS
+
+resource "azurerm_function_app" "azurerm_function_app" {
+  name                      = "${var.azurerm_functionapp}"
+  location                  = "${azurerm_resource_group.azurerm_resource_group.location}"
+  resource_group_name       = "${azurerm_resource_group.azurerm_resource_group.name}"
+  app_service_plan_id       = "${azurerm_app_service_plan.azurerm_app_service_plan.id}"
+  storage_connection_string = "${azurerm_storage_account.azurerm_functionapp_storage_account.primary_connection_string}"
+  version                   = "~1"
+
+  app_settings = {
+    # "AzureWebJobsStorage" = "${azurerm_storage_account.azurerm_functionapp_storage_account.primary_connection_string}"
+    # "AzureWebJobsDashboard" = "${azurerm_storage_account.azurerm_functionapp_storage_account.primary_connection_string}"
+    "COSMOSDB_NAME" = "${var.azurerm_cosmosdb_documentdb}"
+    "QueueStorageConnection" = "${azurerm_storage_account.azurerm_storage_account.primary_connection_string}"
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = "${azurerm_application_insights.azurerm_application_insights.instrumentation_key}"
+    "FUNCTION_APP_EDIT_MODE" = "readonly"
+    "AzureWebJobsSecretStorageType" = "disabled"
```

Comment:
I'd copy the comments for each settings from the old js task, in this case it was 

```
       // AzureWebJobsSecretStorageType may be `disabled` or `Blob`
        // When set to `Blob` the API manager task won't be able
        // to retrieve the master key
```

Reply:
until they fix the problem with connection strings, we can put connection strings in app settings if you prefix them by "CUSTOMCONNSTR" ie.  "CUSTOMCONNSTR_COSMOSDB_URI"=....